File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/user/response/UserResponse.java
package dev.starryeye.book_library.controller.user.response;

import dev.starryeye.book_library.application.command.result.RegisterUserResult;
import dev.starryeye.book_library.domain.User;

public record UserResponse(
        Long id,
        String username,
        Integer age
) {

    public static UserResponse of(RegisterUserResult result) {
        return new UserResponse(
                result.id(),
                result.username(),
                result.age()
        );
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/user/response/UserResponse.kt
package dev.starryeye.book_library.controller.user.response

import dev.starryeye.book_library.application.user.command.result.RegisterUserResult

data class UserResponse(
    val id: Long,
    val username: String,
    val age: Int?,
) {

    companion object {
        fun of(userResult: RegisterUserResult): UserResponse {
            return UserResponse(
                id = userResult.id,
                username = userResult.username,
                age = userResult.age,
            )
        }
    }
}